src/components/Icon/BankIcons.ts

```diff
@@ -52,101 +14,85 @@ type BankIcon = {
+        [BANK_NAMES.DISCOVER]: isCard ? require('@assets/images/cardicons/discover.svg') : require('@assets/images/bankicons/discover.svg'),
+        [BANK_NAMES.FIDELITY]: isCard ? require('@assets/images/cardicons/fidelity.svg') : require('@assets/images/bankicons/fidelity.svg'),
+        [BANK_NAMES.GENERIC_BANK]: isCard ? require('@assets/images/cardicons/generic-bank-card.svg') : require('@assets/images/bankicons/generic-bank-account.svg'),
+        [BANK_NAMES.HUNTINGTON_BANK]: isCard ? require('@assets/images/cardicons/huntington-bank.svg') : require('@assets/images/bankicons/huntington-bank.svg'),
+        [BANK_NAMES.NAVY_FEDERAL_CREDIT_UNION]: isCard
+            ? require('@assets/images/cardicons/navy-federal-credit-union.svg')
+            : require('@assets/images/bankicons/navy-federal-credit-union.svg'),
+        [BANK_NAMES.PNC]: isCard ? require('@assets/images/cardicons/pnc.svg') : require('@assets/images/bankicons/pnc.svg'),
+        [BANK_NAMES.REGIONS_BANK]: isCard ? require('@assets/images/cardicons/regions-bank.svg') : require('@assets/images/bankicons/regions-bank.svg'),
+        [BANK_NAMES.SUNTRUST]: isCard ? require('@assets/images/cardicons/suntrust.svg') : require('@assets/images/bankicons/suntrust.svg'),
+        [BANK_NAMES.TD_BANK]: isCard ? require('@assets/images/cardicons/td-bank.svg') : require('@assets/images/bankicons/td-bank.svg'),
+        [BANK_NAMES.US_BANK]: isCard ? require('@assets/images/cardicons/us-bank.svg') : require('@assets/images/bankicons/us-bank.svg'),
+        [BANK_NAMES.USAA]: isCard ? require('@assets/images/cardicons/usaa.svg') : require('@assets/images/bankicons/usaa.svg'),
+    };
 
-    if (bankName.includes('usaa')) {
-        return isCard ? USAACard : USAA;
-    }
+    // Fallback to generic bank/card icon
+    const iconModule = iconMappings[bankName] || (isCard ? '@assets/images/cardicons/generic-bank-card.svg' : '@assets/images/bankicons/generic-bank-account.svg');
+    return iconModule as React.FC<SvgProps>;
+}
 
-    return isCard ? GenericBankCard : GenericBank;
+function getBankNameKey(bankName: string): BankNameAndEmptyString | undefined {
+    return Object.keys(BANK_NAMES).find((key) => BANK_NAMES[key as BankName].toLowerCase() === bankName) as BankNameAndEmptyString | undefined;
 }
 
 /**
  * Returns Bank Icon Object that matches to existing bank icons or default icons
  */
 
-export default function getBankIcon(bankName: string, isCard = false): BankIcon {
+export default function getBankIcon(bankName: BankNameAndEmptyString, isCard = false): BankIcon {
+    const bankNameKey = getBankNameKey(bankName.toLowerCase());
+
     const bankIcon: BankIcon = {
         icon: isCard ? GenericBankCard : GenericBank,
     };
 
-    if (bankName) {
-        bankIcon.icon = getAssetIcon(bankName.toLowerCase(), isCard);
+    if (bankNameKey && Object.keys(BANK_NAMES).includes(bankNameKey)) {
```

Comment:
Here `bankNameKey` is of type `"EXPENSIFY" | "AMERICAN_EXPRESS" | ...` but icon mappings inside of `getAssetIcon` are defined with `"expensify" | "americanexpress" | ...` so fallback will be used all the time, right?

Reply:
I've just added `CONST.BANK_NAMES[bankNameKey];` to `getAssetIcon` . So now i'm using `"expensify" | "americanexpress" | ...` and it should be good